client.go

```diff
@@ -9,28 +10,52 @@ import (
 // the protocol specified for the handshake.
 var ErrNotSupported = errors.New("protocol not supported")
 
+// ErrNoProtocols is the error returned when the no protocols have been
+// specified.
+var ErrNoProtocols = errors.New("protocol not supported")
```

Comment:
This should be difficult from `ErrNotSupported`, otherwise `ErrNotSupported == ErrNoProtocols`.

Reply:
(I should try to not copy and paste...)